Fixed issues with getting the specific id

elicarcar opened the pull request.

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
```
I don't see `express` in your `package.json`

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const uuidv4 = require('uuid/v4');
```
uuid too!

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+
+function readTodoFiles() {
+  return new Promise(function(resolve, reject) {
+    fs.readFile('todo.json', 'utf8', (err, data) => {
```
instead of this long function you could require it, `.json` files can be required! as:
```javascript
const todos = require('todos')
```

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+
+function validateAndParseTodo(req) {
+  const { todo } = req.body;
+  if (todo == null) throw Error('todo not set');
```
in some case throwing could result in shutting the nodejs server application down!

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+  allData = [];
+  await saveTodos(allData);
+  res.status = 201;
+  res.send();
```
That being said you are not retuning any data on success, I would go for returning some kind of object just as a note to the client saying `Hey your request is succeed`, well it's happening in the status.

but just as a convention you could response with at least `{ ok: 'true' }` which makes it kinda success and same response type.

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+    await saveTodos(todos);
+    res.send();
+  } catch (err) {
+    new Error(console.log(err.message));
```
is this an error or a console, does it send to the client? 🤔

Meno-101 commented on week3/homework/src/index.js:
```diff
@@ -1,3 +1,150 @@
+}
+
+async function deleteAllTodos(req, res) {
+  let allData = await readTodoFiles();
```
This is a correct use of `let` 🎉

elicarcar commented:
Hello @Meno-101 thank you for your feedback! I'm so sorry for implementing them this late. I think i modified all the parts you've mentioned and asked me to change.
So, I added some informative messages to the client on deleteAllTodos function and updateTodo function. I decided to print most of the errors instead of throwing them, installed all the required dependencies and changed all the lets with consts (except the ones that I used in deleteAllTodos function).
